apps/houstonmetro/houstonmetro.star

```diff
@@ -0,0 +1,228 @@
+
+    #Create animation frames of the stop info
+    animation_children = []
+    if len(render_elements) == 1:
+        frame_1 = render.Column(
+            children = [
+                render_elements[0],
+            ],
+        )
+        for i in range(0, 160):
+            animation_children.append(frame_1)
+    if len(render_elements) == 2:
+        frame_1 = render.Column(
+            children = [
+                render_elements[0],
+                render_elements[1],
+            ],
+        )
+        for i in range(0, 160):
+            animation_children.append(frame_1)
+    if len(render_elements) == 3:
+        frame_1 = render.Column(
+            children = [
+                render_elements[0],
+                render_elements[1],
+            ],
+        )
+        frame_2 = render.Column(
+            children = [
+                render_elements[2],
+            ],
+        )
+        for i in range(0, 160):
+            if i <= 80:
+                animation_children.append(frame_1)
+            else:
+                animation_children.append(frame_2)
+    if len(render_elements) == 4:
+        frame_1 = render.Column(
+            children = [
+                render_elements[0],
+                render_elements[1],
+            ],
+        )
+        frame_2 = render.Column(
+            children = [
+                render_elements[2],
+                render_elements[3],
+            ],
+        )
+        for i in range(0, 160):
+            if i <= 80:
+                animation_children.append(frame_1)
+            else:
+                animation_children.append(frame_2)
+
+    return render.Root(
+        child = render.Column(
+            children = [
+                render.Row(
+                    children = [
+                        render.Image(
+                            src = METRO_ICON,
+                        ),
+                        render.Marquee(
+                            child =
+                                render.Text(
+                                    stop_name,
+                                    font = "tb-8",
+                                    height = 12,
+                                ),
+                            align = "center",
+                            width = 45,
+                            offset_start = 5,
+                            offset_end = 32,
+                        ),
+                    ],
+                ),
+                render.Sequence(
+                    children = [
+                        render.Animation(
+                            children = animation_children,
+                        ),
+                    ],
+                ),
+            ],
+        ),
+    )
+
+def time_string(full_string):
+    time_index = full_string.find("T")
+    return full_string[time_index + 1:len(full_string) - 4]
+
+def get_stations(location):
+    loc = json.decode(location)
+    coordinates = str(loc["lat"]) + "|" + str(loc["lng"])
```

Comment:
Would be great to have the coordinates truncated to avoid sharing user's exact location with the API provider. Based on the use case, I would guess 3 decimals for ~100m accuracy is sufficient here?

Check out this post for some additional info on how the number of decimals affect accuracy:
https://gis.stackexchange.com/questions/8650/measuring-accuracy-of-latitude-and-longitude

Reply:
Done!